pyanaconda/modules/localization/localed.py

```diff
@@ -43,6 +47,34 @@ def __init__(self):
             return
 
         self._localed_proxy = LOCALED.get_proxy()
+        self._localed_proxy.PropertiesChanged.connect(self._on_properties_changed)
+
+    def _on_properties_changed(self, interface, changed_props, invalid_props):
+        if "X11Layout" in changed_props or "X11Variant" in changed_props:
+            layouts_variants = self._from_localed_format(changed_props["X11Layout"].get_string(),
+                                                         changed_props["X11Variant"].get_string())
+
+            # layouts has changed in compositor, always emit this signal
+            log.debug("Localed layouts has changed. Last known: '%s' current: '%s'",
+                      self._last_layouts_variants, layouts_variants)
+            self.compositor_layouts_changed.emit(layouts_variants)
+
+            # check if last selected variant has changed
+            # nothing is selected in compositor
+            if not layouts_variants:
+                log.warning("Compositor layouts seems to be not set")
+                self.compositor_selected_layout_changed.emit("")
+            # we don't know last
+            elif not self._last_layouts_variants:
+                log.debug("Missing information about last selected layouts. "
+                          "Use compositor layouts.")
```

Comment:
I'll change that because we are not "using" these layouts. We are just sending signals that this is what compositor is using. The point is that we should be always in control of the layouts.